TheEthicalHackerCup/Assets/Scripts/SecurityConceptProgress.cs

```diff
@@ -25,13 +26,15 @@ public SecurityConceptProgress(int maxDefenseUpgradeLevel = 5)
     public int GetAttackMinigamesAttempted() { return this._attackMinigamesAttempted; }
     public int GetHeat() { return this._heat; }
 
+    public int GetAttackMinigameAttemptsRequiredToUpgrade() { return Math.Min(this._maxDefenseUpgradeLevel, this._currentDefenseUpgradeLevel + 1); }
+
     // Boolean Indicators
     public bool IsFullyUpgraded() { return this._currentDefenseUpgradeLevel >= this._maxDefenseUpgradeLevel; }
 
     // Boosters
     public bool UpgradeDefense()
     {
-        if (this._currentDefenseUpgradeLevel + 1 > this._maxDefenseUpgradeLevel)
+        if (this._currentDefenseUpgradeLevel + 1 > this._maxDefenseUpgradeLevel || this._attackMinigamesAttempted < this.GetAttackMinigameAttemptsRequiredToUpgrade())
```

Comment:
Why this + 1 instead of >=?

Reply:
Originally, it felt more natural when writing it out to say  "When attempting to upgrade, compare the currentLevel+1 to the max"

I replaced this with >= to save a the single assembly instruction required to add 1. Good point